datacube/utils/geometry/_base.py

```diff
@@ -171,20 +174,20 @@ def __init__(self, crs_spec: Any):
         elif isinstance(crs_spec, dict):
             self._crs, self._str, self._epsg = _make_crs(_CRS.from_dict(crs_spec))
         else:
-            try:
-                epsg = crs_spec.to_epsg()
-            except AttributeError:
-                epsg = None
-            if epsg is not None:
-                self._crs, self._str, self._epsg = _make_crs(f"EPSG:{epsg}")
-                return
             try:
                 wkt = crs_spec.to_wkt()
             except AttributeError:
                 wkt = None
             if wkt is not None:
                 self._crs, self._str, self._epsg = _make_crs(wkt)
                 return
+            try:
+                epsg = crs_spec.to_epsg()
```

Comment:
Use EPSG as backup to WKT. Likely never used. Will raise exception in the future if no WKT.